featurebyte/service/observation_table.py

```diff
@@ -359,27 +363,77 @@ async def update_observation_table(
         ------
         ObservationTableInvalidContextError
             If the entity ids are different from the existing Context
+        ObservationTableInvalidUseCaseError
+            If the use case is invalid
         """
 
+        if data.use_case_ids:
+            raise ObservationTableInvalidUseCaseError("use_case_ids is not a valid field to update")
+
+        observation_table = await self.get_document(document_id=observation_table_id)
+        if data.use_case_id_to_add or data.use_case_id_to_remove:
+            if not observation_table.context_id:
+                raise ObservationTableInvalidUseCaseError(
+                    f"Cannot add/remove UseCase as the ObservationTable {observation_table_id} is not associated with any Context."
+                )
```

Comment:
Would be good to have an unit test to cover this.

Reply:
The test is here: https://github.com/featurebyte/featurebyte/blob/yanhui/DEV-2435_2/tests/unit/routes/test_observation_table.py#L263